components/ws-daemon/nsinsider/main.go

```diff
@@ -254,6 +258,193 @@ func main() {
 					return nil
 				},
 			},
+			{
+				Name:  "setup-pair-veths",
+				Usage: "set up a pair of veths",
+				Flags: []cli.Flag{
+					&cli.IntFlag{
+						Name:     "target-pid",
+						Required: true,
+					},
+				},
+				Action: func(c *cli.Context) error {
+					containerIf, vethIf, cethIf := "eth0", "veth0", "ceth0"
+					mask := net.IPv4Mask(255, 255, 255, 0)
+					vethIp := net.IPNet{
+						IP:   net.IPv4(10, 0, 5, 1),
+						Mask: mask,
+					}
+					masqueradeAddr := net.IPNet{
+						IP:   vethIp.IP.Mask(mask),
+						Mask: mask,
+					}
+					// iptablesCmd := "/usr/sbin/iptables"
+					targetPid := c.Int("target-pid")
+
+					veth := &netlink.Veth{
+						LinkAttrs: netlink.LinkAttrs{
+							Name:  vethIf,
+							Flags: net.FlagUp,
+							MTU:   1500,
+						},
+						PeerName:      cethIf,
+						PeerNamespace: netlink.NsPid(targetPid),
+					}
+					if err := netlink.LinkAdd(veth); err != nil {
+						return xerrors.Errorf("link %q-%q netns failed: %v", vethIf, cethIf, err)
+					}
+
+					vethLink, err := netlink.LinkByName(vethIf)
+					if err != nil {
+						return xerrors.Errorf("cannot found %q netns failed: %v", vethIf, err)
+					}
+					if err := netlink.AddrAdd(vethLink, &netlink.Addr{IPNet: &vethIp}); err != nil {
+						return xerrors.Errorf("failed to add IP address to %q: %v", vethIf, err)
+					}
+					if err := netlink.LinkSetUp(vethLink); err != nil {
+						return xerrors.Errorf("failed to enable %q: %v", vethIf, err)
+					}
+
+					nc := &nftables.Conn{}
+					nat := nc.AddTable(&nftables.Table{
+						Family: nftables.TableFamilyIPv4,
+						Name:   "nat",
+					})
+
+					postrouting := nc.AddChain(&nftables.Chain{
+						Name:     "postrouting",
+						Hooknum:  nftables.ChainHookPostrouting,
+						Priority: nftables.ChainPriorityNATSource,
+						Table:    nat,
+						Type:     nftables.ChainTypeNAT,
+					})
+
+					// ip saddr 10.0.5.0/24 oifname "eth0" masquerade
+					nc.AddRule(&nftables.Rule{
```

Comment:
If you are interested in the rule commands, I would recommend the following resources. However, these may be a bit tougher to understand. If you have experience with eBPF, maybe you are familiar with these rule commands, because I feel it is similar to eBPF.
- [Quick reference-nftables in 10_minutes](https://wiki.nftables.org/wiki-nftables/index.php/Quick_reference-nftables_in_10_minutes)
- [test codes of google/nftables](https://github.com/google/nftables/blob/master/nftables_test.go)